test/e2e/operator/handler_install_test.go

```diff
@@ -0,0 +1,91 @@
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	nmstate       nmstatev1beta1.NMState
+	nmstateKey    = types.NamespacedName{Namespace: "nmstate", Name: "nmstate"}
+	dsKey         = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-handler"}
+	deploymentKey = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-operator"}
+)
+
+func installHandler() {
+	err := framework.Global.Client.Get(context.TODO(), nmstateKey, &nmstate)
+	Expect(err).To(HaveOccurred())
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).To(Not(HaveOccurred()))
+}
+
+func uninstallHandler() {
+	err := framework.Global.Client.Get(context.TODO(), nmstateKey, &nmstate)
+	Expect(err).To(Not(HaveOccurred()))
+	err = framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+	Expect(err).To(Not(HaveOccurred()))
+}
+
+var _ = Describe("handler install", func() {
+	BeforeEach(func() {
+		nmstate = nmstatev1beta1.NMState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nmstate",
+				UID:  "12345",
+			},
+		}
+	})
+	Context("when handler not installed", func() {
+		It("should install", func() {
+			installHandler()
+			Eventually(func() string {
+				ds := v1.DaemonSet{}
+				_ = framework.Global.Client.Get(context.TODO(), dsKey, &ds)
+				return ds.ObjectMeta.Name
+			}, 60*time.Second, 1*time.Second).ShouldNot(BeEmpty(), fmt.Sprintf("Could not find daemon set for handler"))
+		})
+	})
+	Context("when handler is installed", func() {
+		It("should ignore a CR of the same name", func() {
+			nmstateList := nmstatev1beta1.NMStateList{}
+			err := framework.Global.Client.List(context.TODO(), &nmstateList, &client.ListOptions{})
+			Expect(err).To(Not(HaveOccurred()))
+			Expect(nmstateList.Items).To(HaveLen(1))
+			err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+			Expect(nmstateList.Items).To(HaveLen(1))
+		})
+		It("should ignore a CR of the wrong name", func() {
+			defer func() {
+				nmstate.SetName("nmstate-wrong")
+				framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+			}()
+			ds := v1.DaemonSetList{}
+			err := framework.Global.Client.List(context.TODO(), &ds, &client.ListOptions{Namespace: "nmstate"})
+			Expect(err).To(Not(HaveOccurred()))
+			Expect(ds.Items).To(HaveLen(1))
+			nmstate.SetName("nmstate-wrong")
+			err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+			Expect(err).To(Not(HaveOccurred()))
+			err = framework.Global.Client.List(context.TODO(), &ds, &client.ListOptions{Namespace: "nmstate"})
+			Expect(ds.Items).To(HaveLen(1))
```

Comment:
This is suppose to be HaveLen(2) the two CR are there but it's ignored I think, to really check that it's ignored we can add a label or something to the CR at controller or check back handler/webhook.

Reply:
That's checking the DaemonSetList, not the NMStatesList. That would have 2 CRs. This is just checking that another handler DaemonSet doesn't get deployed.